include/tscore/BaseLogFile.h

```diff
@@ -99,6 +99,18 @@ class BaseMetaInfo
 
   void _read_from_file();
   void _write_to_file();
+
+  /** Repeat write calls to fd until all len bytes are written.
+   *
+   * @param[in] fd The file descriptor to write to.
+   * @param[in] buffer The buffer of bytes to write into fd.
+   * @param[in] len The number of bytes in buffer to write into fd.
+   *
+   * @return The number of bytes written or -1 if there was an error writing to
+   * the file descriptor. If -1 is returned, errno should indicate why the
+   * write failed.
+   */
+  int _write_till_buffer_is_drained(int fd, char const *buffer, int len);
```

Comment:
Alan suggested offline adding a `safe_write` method. That's a good idea, so I'll look into that.